lib/services/android-project-service.ts

```diff
@@ -182,104 +190,61 @@ class AndroidProjectService implements IPlatformProjectService {
 		}
-
-		let lines = this.$fs.readText(projProp, "utf-8").wait().split(os.EOL);
-
-		let regEx = /android\.library\.reference\.(\d+)=(.*)/;
-		lines.forEach(elem => {
-			let match = elem.match(regEx);
-			if (match) {
-				let libRef: ILibRef = { idx: parseInt(match[1]), path: match[2].trim() };
-				libRef.adjustedPath = this.$fs.isRelativePath(libRef.path) ? path.join(projDir, libRef.path) : libRef.path;
-				this.parseProjectProperties(libRef.adjustedPath, destDir);
-			}
-		});
-
-		this.$logger.info("Copying %s", projDir);
-		shell.cp("-Rf", projDir, destDir);
-
-		let targetDir = path.join(destDir, path.basename(projDir));
-		// TODO: parametrize targetSdk
-		let targetSdk = "android-17";
-		this.$logger.info("Generate build.xml for %s", targetDir);
-		this.runAndroidUpdate(targetDir, targetSdk).wait();
-	}
-
-	private getProjectReferences(projDir: string): ILibRef[]{
-		let projProp = path.join(projDir, "project.properties");
-
-		let lines = this.$fs.readText(projProp, "utf-8").wait().split(os.EOL);
-
-		let refs: ILibRef[] = [];
-
-		let regEx = /android\.library\.reference\.(\d+)=(.*)/;
-		lines.forEach(elem => {
-			let match = elem.match(regEx);
-			if (match) {
-				let libRef: ILibRef = { idx: parseInt(match[1]), path: match[2] };
-				libRef.adjustedPath = path.join(projDir, libRef.path);
-				refs.push(libRef);
-			}
-		});
-
-		return refs;
+		
+		return this._androidProjectPropertiesManagers[filePath];
 	}
-
-	private updateProjectReferences(projDir: string, libraryPath: string): void {
-		let refs = this.getProjectReferences(projDir);
-		let maxIdx = refs.length > 0 ? _.max(refs, r => r.idx).idx : 0;
-
-		let relLibDir = path.relative(projDir, libraryPath).split("\\").join("/");
-
-		let libRefExists = _.any(refs, r => path.normalize(r.path) === path.normalize(relLibDir));
-
-		if (!libRefExists) {
-			let projRef = util.format("%sandroid.library.reference.%d=%s", os.EOL, maxIdx + 1, relLibDir);
+	
+	private parseProjectProperties(projDir: string, destDir: string): IFuture<void> { // projDir is libraryPath, targetPath is the path to lib folder
+		return (() => {
 			let projProp = path.join(projDir, "project.properties");
-			fs.appendFileSync(projProp, projRef, { encoding: "utf-8" });
-		}
+			if (!this.$fs.exists(projProp).wait()) {
+				this.$logger.warn("Warning: File %s does not exist", projProp);
+				return;
+			}
+			
+			let projectPropertiesManager = this.getProjectPropertiesManager(projDir);
+			let references = projectPropertiesManager.getProjectReferences().wait();
+			_.each(references, reference => {
+				let adjustedPath = this.$fs.isRelativePath(reference.path) ? path.join(projDir, reference.path) : reference.path;
+				this.parseProjectProperties(adjustedPath, destDir).wait();
+			});
+	
+			this.$logger.info("Copying %s", projDir);
+			shell.cp("-Rf", projDir, destDir);
+	
+			let targetDir = path.join(destDir, path.basename(projDir));
+			let targetSdk = `android-${this.$options.sdk || 17}`;
```

Comment:
Can you set 17 as a constant, so we can easily search for it later (when we decide to support later version).